library/spdm_transport_tcp_lib/libspdm_tcp_tcp.c

```diff
@@ -0,0 +1,219 @@
+ *
+ * This value is transport layer specific.
+ *
+ * @return Max random number count in an SPDM secured message.
+ *        0 means no randum number is required.
+ **/
+uint32_t libspdm_tcp_get_max_random_number_count(void)
+{
+    return LIBSPDM_TCP_MAX_RANDOM_NUMBER_COUNT;
+}
+
+/**
+ * This function translates the negotiated secured_message_version to a DSP0277 version.
+ *
+ * @param  secured_message_version  The version specified in binding specification and
+ *                                  negotiated in KEY_EXCHANGE/KEY_EXCHANGE_RSP.
+ *
+ * @return The DSP0277 version specified in binding specification,
+ *         which is bound to secured_message_version.
+ */
+spdm_version_number_t libspdm_tcp_get_secured_spdm_version(
+    spdm_version_number_t secured_message_version)
+{
+    return secured_message_version;
+}
+
+/**
+ * Encode a normal message or secured message to a transport message.
+ *
+ * @param  session_id                    Indicates if it is a secured message protected via SPDM session.
+ *                                     If session_id is NULL, it is a normal message.
+ *                                     If session_id is NOT NULL, it is a secured message.
+ * @param  message_size                  size in bytes of the message data buffer.
+ * @param  message                      A pointer to a source buffer to store the message.
+ * @param  transport_message_size         size in bytes of the transport message data buffer.
+ * @param  transport_message             A pointer to a destination buffer to store the transport message.
+ *
+ * @retval RETURN_SUCCESS               The message is encoded successfully.
+ * @retval RETURN_INVALID_PARAMETER     The message is NULL or the message_size is zero.
+ **/
+
+libspdm_return_t libspdm_tcp_encode_message(const uint32_t *session_id,
+                                            size_t message_size,
+                                            void *message,
+                                            size_t *transport_message_size,
+                                            void **transport_message)
+{
+    size_t alignment = LIBSPDM_TCP_ALIGNMENT;
+    size_t aligned_message_size = (message_size + (alignment - 1)) & ~(alignment - 1);
+    size_t required_size;
+    tcp_spdm_binding_header_t *tcp_message_header;
+    uint8_t *buffer;
+
+    /* Determine required buffer size */
+    required_size = (session_id == NULL && message == NULL && message_size == 0) ?
+                    sizeof(tcp_spdm_binding_header_t) :
+                    sizeof(tcp_spdm_binding_header_t) + aligned_message_size;
+
+    /* Check buffer size */
+    if (*transport_message_size < required_size) {
+        *transport_message_size = required_size;
+        return LIBSPDM_STATUS_BUFFER_TOO_SMALL;
+    }
+
+    *transport_message_size = required_size;
+    buffer = (uint8_t *)*transport_message;
+
+    if (session_id == NULL && message == NULL && message_size == 0) {
```

Comment:
This is a brittle way to encode this information. In particular there may be other `MessageType`s introduced. I'll need to think about this one. @jyao1 may want to think too.

Reply:
Sure, as suggested in the Code TF meeting, I have removed the Role Inquiry message from this PR. That will be added as part of another PR later.